test/integration/targets/azure_rm_gallery/tasks/main.yml

```diff
@@ -0,0 +1,251 @@
+  azure_rm_galleryimage:
+    resource_group: "{{ resource_group }}"
+    gallery_name: myGallery{{ rpfx }}
+    name: myImage
+    location: West US
+    os_type: linux
+    os_state: generalized
+    identifier:
+      publisher: myPublisherName
+      offer: myOfferName
+      sku: mySkuName
+    description: Image Description XXXs
+  register: output
+
+- assert:
+    that: 
+      - output.changed
+
+- name: Create or update a simple gallery Image Version.
+  azure_rm_galleryimageversion:
+    resource_group: "{{ resource_group }}"
+    gallery_name: myGallery{{ rpfx }}
+    gallery_image_name: myImage
+    name: 10.1.3
+    location: West US
+    publishing_profile:
+      end_of_life_date: "2020-10-01t00:00:00+00:00"
+      exclude_from_latest: yes
+      replica_count: 3
+      storage_account_type: Standard_LRS
+      target_regions:
+        - name: West US
+          regional_replica_count: 1
+        - name: East US
+          regional_replica_count: 2
+          storage_account_type: Standard_ZRS
+      managed_image:
+        name: testimagea
+        resource_group: "{{ resource_group }}"
+  register: output
+
+- assert:
+    that: 
+      - output.changed
+
+- name: Create or update a simple gallery Image Version - idempotent
+  azure_rm_galleryimageversion:
+    resource_group: "{{ resource_group }}"
+    gallery_name: myGallery{{ rpfx }}
+    gallery_image_name: myImage
+    name: 10.1.3
+    location: West US
+    publishing_profile:
+      end_of_life_date: "2020-10-01t00:00:00+00:00"
+      exclude_from_latest: yes
+      replica_count: 3
+      storage_account_type: Standard_LRS
+      target_regions:
+        - name: West US
+          regional_replica_count: 1
+        - name: East US
+          regional_replica_count: 2
+          storage_account_type: Standard_ZRS
+      managed_image:
+        name: testimagea
+        resource_group: "{{ resource_group }}"
+  register: output
+
+- assert:
+    that: 
+      - not output.changed
+
+- name: Create or update a simple gallery Image Version - change end of life
+  azure_rm_galleryimageversion:
+    resource_group: "{{ resource_group }}"
+    gallery_name: myGallery{{ rpfx }}
+    gallery_image_name: myImage
+    name: 10.1.3
+    location: West US
+    publishing_profile:
+      end_of_life_date: "2021-10-01t00:00:00+00:00"
+      exclude_from_latest: yes
+      replica_count: 3
+      storage_account_type: Standard_LRS
+      target_regions:
+        - name: West US
+          regional_replica_count: 1
+        - name: East US
+          regional_replica_count: 2
+          storage_account_type: Standard_ZRS
+      managed_image:
+        name: testimagea
+        resource_group: "{{ resource_group }}"
+  register: output
+
+- assert:
+    that: 
+      - output.changed
+
+# do  not test delete at the moment as it will fail due to mysterious nested resource
+# deleting resource group will delete all the resources
+
+#- name: Delete gallery Image Version.
```

Comment:
Does deletion work without replication? I'm wondering if it's preventing deletion because the replication is in progress...